src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -199,7 +215,14 @@ class RegularSync(
     case ResponseReceived(peer, BlockBodies(blockBodies), timeTaken) =>
       log.info("Received {} block bodies in {} ms", blockBodies.size, timeTaken)
       waitingForActor = None
-      handleBlockBodies(peer, blockBodies)
+      matchBlockBodiesWithHeaders(peer, blockBodies).foreach { blocks =>
+        handleBlocks(peer, blocks)
+      }
+
+    case ResponseReceived(peer, NodeData(nodes), timeTaken) =>
```

Comment:
maybe add `if missingStateNodeRetry.isDefined` (it'll solve my comment below about using .get).

Reply:
Good catch